### PR TITLE
Add support for jdk12 to pipeline variables

### PR DESCRIPTION
**Pull request by AdamBrousseau**
[skip ci]
Issue #4065

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Perhaps a good time to reconsider #2016

File: buildenv/jenkins/variables/defaults.yml [hunk 1]
```diff
@@ -68,6 +73,7 @@ build_discarder:
 extra_getsource_options:
   8: '--openssl-version=1.1.1'
   11: '--openssl-version=1.1.1'
+  12: '--openssl-version=1.1.1'
```
Comment (pshipton) [inline buildenv/jenkins/variables/defaults.yml]: We don't need this yet. 12 is the same as next and not 11.

File: buildenv/jenkins/variables/defaults.yml [hunk 2]
```diff
@@ -80,12 +86,14 @@ linux_ppc-64_cmprssptrs_le:
...
     10: '/usr/lib/jvm/adoptojdk-java-ppc64le-90'
     11: '/usr/lib/jvm/adoptojdk-java-11'
+    12: '/usr/lib/jvm/adoptojdk-java-11'
     next: '/usr/lib/jvm/adoptojdk-java-11'
   release:
     8: 'linux-ppc64le-normal-server-release'
     9: 'linux-ppc64le-normal-server-release'
     10: 'linux-ppc64le-normal-server-release'
     11: 'linux-ppc64le-normal-server-release'
+    12: 'linux-ppc64le-normal-server-release'
```
Comment (pshipton) [inline buildenv/jenkins/variables/defaults.yml]: 12 is the same as next, not 11.

File: buildenv/jenkins/variables/defaults.yml [hunk 3]
```diff
@@ -410,25 +463,30 @@ osx_x86-64:
...
     next: '/Users/jenkins/bootjdks/adoptojdk-java-11'
   release:
     8: 'macosx-x86_64-normal-server-release'
     11: 'macosx-x86_64-normal-server-release'
+    12: 'macosx-x86_64-normal-server-release'
     next: 'macosx-x86_64-server-release'
   extra_configure_options:
     8: '--with-noncompressedrefs --with-xcode-path=/Users/jenkins/Xcode4/Xcode.app --with-openj9-cc=/Users/jenkins/Xcode7/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang --with-openj9-cxx=/Users/jenkins/Xcode7/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ --with-openj9-developer-dir=/Users/jenkins/Xcode7/Xcode.app/Contents/Developer'
     11: '--with-noncompressedrefs --with-openssl=fetched --enable-openssl-bundling'
+    12: '--with-noncompressedrefs --with-openssl=fetched --enable-openssl-bundling'
```
Comment (pshipton) [inline buildenv/jenkins/variables/defaults.yml]: I must have merged something incorrectly. 12 and next do not support openssl yet and should not have openssl parameters. This applies to all platforms.

**Comment by AdamBrousseau** [comment]
Updated.

File: buildenv/jenkins/variables/defaults.yml [hunk 4]
```diff
@@ -410,25 +458,28 @@ osx_x86-64:
...
     next: '/Users/jenkins/bootjdks/adoptojdk-java-11'
   release:
     8: 'macosx-x86_64-normal-server-release'
     11: 'macosx-x86_64-normal-server-release'
+    12: 'macosx-x86_64-server-release'
     next: 'macosx-x86_64-server-release'
   extra_configure_options:
     8: '--with-noncompressedrefs --with-xcode-path=/Users/jenkins/Xcode4/Xcode.app --with-openj9-cc=/Users/jenkins/Xcode7/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang --with-openj9-cxx=/Users/jenkins/Xcode7/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ --with-openj9-developer-dir=/Users/jenkins/Xcode7/Xcode.app/Contents/Developer'
     11: '--with-noncompressedrefs --with-openssl=fetched --enable-openssl-bundling'
-    next: '--with-noncompressedrefs --with-openssl=fetched --enable-openssl-bundling'
```
Comment (pshipton) [inline buildenv/jenkins/variables/defaults.yml]: The `--with-noncompressedrefs` part is still needed

File: buildenv/jenkins/variables/defaults.yml [hunk 5]
```diff
@@ -410,25 +458,28 @@ osx_x86-64:
...
   node_labels:
     build:
       8: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_11'
       11: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_13'
+      12: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_13'
       next: 'ci.role.build && hw.arch.x86 && sw.os.osx.10_13'
     test:
       8: 'hw.arch.x86 && sw.os.osx'
       11: 'hw.arch.x86 && sw.os.osx'
+      12: 'hw.arch.x86 && sw.os.osx'
```
Comment (pshipton) [inline buildenv/jenkins/variables/defaults.yml]: Why no "next:" here?

**Pull request merged**